### PR TITLE
Port Fline_beginning_position and Fline_end_position

### PR DESCRIPTION
**Pull request by shaleh**
Port Ffield_begin and Ffield_end Port Fconstrain_to_field Port
Fget_char_property

### CODE REVIEW

File: rust_src/src/editfns.rs [hunk 1]
```diff
@@ -341,4 +347,242 @@ pub fn string_to_char(string: LispStringRef) -> EmacsInt {
...
+    constrain_to_field(
+        Some(LispNumber::Fixnum(end_pos as EmacsInt)),
+        LispNumber::Fixnum(orig),
+        n != 1,
+        true,
+        LispObject::constant_nil(),
+    )
+}
+
+#[lisp_fn(min = "0")]
```
Comment (Wilfred) [inline rust_src/src/editfns.rs]: You're missing docstrings on a couple of these.

Comment (shaleh) [reply rust_src/src/editfns.rs]: Oh. I was switching functions during debugging and must have neglected to return the comments. Easy fix.

**Comment by shaleh** [comment]
Mac tests floundering. One more time

**Comment by shaleh** [comment]
Mac tests still floundering...

**Comment by shaleh** [comment]
Bah, now windows gives up....

**Comment by shaleh** [comment]
And back to Mac failing....

**Comment by shaleh** [comment]
Back to windows.....

**Comment by shaleh** [comment]
And still windows....

**Comment by Wilfred** [comment]
We should disable these tests on OS X and Windows. Flaky tests are no fun when you have a legitimate PR, especially since the build isn't very quick.

**Comment by shaleh** [comment]
I am loathe to disable them entirely. Once a PR is clearly ok and it is just flaky tests we can manually import and push the change to the Wilfred/remacs master line.

**Comment by shaleh** [comment]
Or not. I just tried to do a manual push. No joy. The check for approved && tested blocks it.

**Comment by Wilfred** [comment]
Agreed, let's disable any test that isn't reliable.

**Comment by shaleh** [comment]
Agreed. I am going to submit a PR that disables OSX.

**Pull request merged**